omniscidb/QueryEngine/Descriptors/QueryFragmentDescriptor.h

```diff
@@ -100,6 +100,56 @@ class QueryFragmentDescriptor {
     }
   }
 
+  template <typename DISPATCH_FCN>
+  void assignFragsToMultiHeterogeneousDispatch(
```

Comment:
Having a separate method is a bit ugly (that was the reason we didn't merge it long time ago), but I think we can live with that for now.